qcfractal/interface/collections/collection.py

```diff
@@ -33,19 +35,34 @@ def __init__(self, name, **kwargs):
         if (self.client is not None) and not (self.client.__class__.__name__ == "FractalClient"):
             raise TypeError("Expected FractalClient as `client` kwarg, found {}.".format(type(self.client)))
 
-        # Init from raw json blob, ignore everything else
-        if kwargs.get("json_data", False):
-            self.data = kwargs["json_data"]
+        if 'collection' not in kwargs:
+            kwargs['collection'] = self.__class__.__name__.lower()
 
-        # Base init
-        else:
-            class_name = self.__class__.__name__.lower()
-            self.data = {
-                "name": name,
-                "collection": class_name,
-                "provenance": kwargs.pop("provenance", {}),
-                **self._init_collection_data(kwargs)
-            }
+        kwargs['name'] = name
+
+        # Create the data model
+        self.data = self.DataModel(**kwargs)
+
+    class DataModel(BaseModel):
+        """
+        Internal Data structure base model typed by PyDantic
+
+        This structure validates input, allows server-side validation and data security,
+        and will create the information to pass back and forth between server and client
+
+        Subclasses of Collection can extend this class internally to change the set of
+        additional data defined by the Collection
+        """
+        name: str
+        collection: str = None
+        provenance: Dict[str, str] = {}
+        id: str = 'local'
```

Comment:
@doaa-altarawy Is there a better thing here than `id` so that we do not get clashes with Python namespace? Not terribly worried about it and keeps it nice and consistent throughout the entire stack, but thought I would ask.